src/main/asciidoc/project/jts/trailmap.adoc

```diff
@@ -0,0 +1,4765 @@
+This is usually done in the form of transaction context.
+
+Transaction context is an association between the transactional operations on the resources, and the components invoking the operations.
+During the course of a transaction, all the threads participating in the transaction share the transaction context.
+Thus the transaction context logically envelops all the operations performed on transactional resources during a transaction.
+The transaction context is usually maintained transparently by the underlying transaction manager.
+
+===== Resource Enlistment
+
+Resource enlistment is the process by which resource managers inform the transaction manager of their participation in a transaction.
+This process enables the transaction manager to keep track of all the resources participating in a transaction.
+The transaction manager uses this information to coordinate transactional work performed by the resource managers and to drive two-phase and recovery protocol.
+At the end of a transaction (after a commit or rollback) the transaction manager delists the resources.
+
+===== Two-Phase Commit
+
+This protocol between the transaction manager and all the resources enlisted for a transaction ensures that either all the resource managers commit the transaction or they all abort.
+In this protocol, when the application requests for committing the transaction, the transaction manager issues a prepare request to all the resource managers involved.
+Each of these resources may in turn send a reply indicating whether it is ready for commit or not.
+Only The transaction manager issues a commit request to all the resource managers, only when all the resource managers are ready for a commit.
+Otherwise, the transaction manager issues a rollback request and the transaction will be rolled back.
+
+===== Recovery and Logging
+
+Basically, the Recovery is the mechanism which preserves the transaction atomicity in presence of failures.
+The basic technique for implementing transactions in presence of failures is based on the use of logs.
+That is, a transaction system has to record enough information to ensure that it can be able to return to a previous state in case of failure or to ensure that changes committed by a transaction are properly stored.
+
+In addition to be able to store appropriate information, all participants within a distributed transaction must log similar information, which allows them to take a same decision either to set data in their final state or in their initial state.
+
+Two techniques are, in general, used to ensure transaction's atomicity.
+A first technique focuses on manipulated data, such as the Do/Undo/Redo protocol (considered as a recovery mechanism in a centralised system), which allows a participant to set its data in their final values or to retrieve them in their initial values.
```

Comment:
This file is strangely long! I can see that it is well migrated, but I can see duplication in this new file and the old one. i.e. these lines https://github.com/jbosstm/documentation/pull/73/files#diff-62ade1ca747887147020c9aa09a6bcdf9074ceb197e58ec5cd41cc67eecb1629R278-R333 can be found also at

Reply:
Thanks @marcosgopen. I also noticed that `trailmap.adoc` has some duplications (and other problems), which are also present in the current documentation. My suggestion is to go ahead with the review and then fix `trailmap.adoc` once we merge this PR. WDYT?